### DISCUSSION TITLE
Details on how to build on Windows? Missing dependencies...

### DISCUSSION THREAD

**Discussion by GuangSunshine** [PintaProject/Pinta#630] [Q&A] [answered]
Hi all, I cloned the Pinta code and tried to build it. The readme.md simply says "opening Pinta.sln in [Visual Studio](https://visualstudio.microsoft.com/). Ensure that .NET 8 is installed", but I got many errors on missing DLLs when building the Pinta project: 

> error MSB3030: Could not copy the file "C:\msys64\mingw64\bin\libbz2-1.dll" because it was not found.
> error MSB3030: Could not copy the file "C:\msys64\mingw64\bin\libbrotlicommon.dll" because it was not found.
> error MSB3030: Could not copy the file "C:\msys64\mingw64\bin\libcairo-gobject-2.dll" because it was not found.
> error MSB3030: Could not copy the file "C:\msys64\mingw64\bin\libcairo-script-interpreter-2.dll" because it was not found.
> error MSB3030: Could not copy the file "C:\msys64\mingw64\bin\libcrypto-3-x64.dll" because it was not found.
> error MSB3030: Could not copy the file "C:\msys64\mingw64\bin\libappstream-5.dll" because it was not found.
> error MSB3030: Could not copy the file "C:\msys64\mingw64\bin\libbrotlidec.dll" because it was not found.
> ...
> error MSB3030: Could not copy the file "C:\msys64\mingw64\bin\libzstd.dll" because it was not found.
> error MSB3030: Could not copy the file "C:\msys64\mingw64\bin\zlib1.dll" because it was not found.
> error MSB3030: Could not copy the file "C:\msys64\mingw64\bin\gdbus.exe" because it was not found.
> error MSB3030: Could not copy the file "C:\msys64\mingw64\bin\gdk-pixbuf-query-loaders.exe" because it was not found.
> error MSB3030: Could not copy the file "C:\msys64\mingw64\bin\gspawn-win64-helper.exe" because it was not found.
> error MSB3030: Could not copy the file "C:\msys64\mingw64\bin\gspawn-win64-helper-console.exe" because it was not found.
> error MSB3030: Could not copy the file "C:\msys64\mingw64\bin\gtk4-update-icon-cache.exe" because it was not found.
> error MSB3030: Could not copy the file "C:\msys64\mingw64\bin\gtk4-query-settings.exe" because it was not found.
> 

I have MINGW64 installed, I have "C:\msys64\mingw64\bin\" directory but it is empty. Looks like I am missing a lot of dependencies, can someone help to list out what libraries/SDKs need to be downloaded and installed? Thanks! And I wish the readme.md would include more details!

**Comment by cameronwhite** [comment]
Sorry about that! I'll get the readme updated shortly
Pinta's master branch has been ported to GTK4, but I only got it working on Windows a week or two ago and hadn't updated the readme yet

I think all you should need is to install the `libadwaita` package (`mingw-w64-x86_64-libadwaita`)
The CI build has the official steps used to generate the installer:
https://github.com/PintaProject/Pinta/blob/f2498065cbdc29121d06bf353493f37e6bdbc5a1/.github/workflows/build.yml#L127

Let me know if that gets it working for you

**Reply by GuangSunshine** [reply, marked as answer]
Thanks for the quick response on Christmas Eve! I installed the `libadwaita` package and Pinta builds and runs! 💯 

Thank you! Merry Christmas and Happy New Year!

PS: To install the `libadwaita` package, I started my MSYS2 MINGW64 Shell and typed `pacman -S mingw-w64-x86_64-libadwaita
`, then I can build in VS!

**Reply by cameronwhite** [reply]
Great, thanks for confirming, and happy holidays!

**Reply by kyriacosmichael** [reply]
In my case i also needed to run this
pacman -S mingw-w64-x86_64-nghttp3
as C:\msys64\mingw64\bin\libnghttp3-9.dll was missing

**Reply by cameronwhite** [reply]
That appeared pretty recently as a new dependency of `cairo` (https://github.com/PintaProject/Pinta/commit/02c1de956df45726f6d0fd441f97bc7d3c4d3f5c), so perhaps your existing mingw packages were out of date?

**Comment by IArnaut2** [comment]
I have installed MSYS2 with Clang64 in `C:/_dev` and I also get this error. How do I make the project use `C:\_dev\msys64\clang64\bin`?
